Index: zfs-linux-0.6.4.2/module/zfs/vdev_disk.c
===================================================================
--- zfs-linux-0.6.4.2.orig/module/zfs/vdev_disk.c
+++ zfs-linux-0.6.4.2/module/zfs/vdev_disk.c
@@ -433,7 +433,7 @@ BIO_END_IO_PROTO(vdev_disk_physio_comple
 		    "bi_idx: %d, bi_size: %d, bi_end_io: %p, bi_cnt: %d\n",
 		    bio->bi_next, bio->bi_flags, bio->bi_rw, bio->bi_vcnt,
 		    BIO_BI_IDX(bio), BIO_BI_SIZE(bio), bio->bi_end_io,
-		    atomic_read(&bio->bi_cnt));
+		    atomic_read(&bio->__bi_cnt));
 
 #ifndef HAVE_2ARGS_BIO_END_IO_T
 	if (BIO_BI_SIZE(bio))
