Index: zfs-linux-0.6.5.3/lib/libspl/include/sys/types.h
===================================================================
--- zfs-linux-0.6.5.3.orig/lib/libspl/include/sys/types.h	2015-10-27 14:38:02.916607000 -0600
+++ zfs-linux-0.6.5.3/lib/libspl/include/sys/types.h	2015-10-27 14:38:23.032607499 -0600
@@ -30,7 +30,6 @@
 #include <sys/isa_defs.h>
 #include <sys/feature_tests.h>
 #include_next <sys/types.h>
-#include <sys/param.h> /* for NBBY */
 #include <sys/types32.h>
 #include <sys/va_list.h>
 
@@ -96,4 +95,6 @@
 } lloff_t;
 #endif
 
+#include <sys/param.h> /* for NBBY */
+
 #endif
