Index: zfs/etc/systemd/system/zfs-import-cache.service.in
===================================================================
--- zfs.orig/etc/systemd/system/zfs-import-cache.service.in
+++ zfs/etc/systemd/system/zfs-import-cache.service.in
@@ -8,7 +8,6 @@ After=systemd-udev-settle.service
 After=systemd-remount-fs.service
 Before=dracut-mount.service
 Before=zfs-import.target
-ConditionPathExists=@sysconfdir@/zfs/zpool.cache
 
 [Service]
 Type=oneshot
